<?xml version="1.0"?>

<launch>

<!-- Audibot launch -->
  <include file="$(find ugv_course_launch)/launch/spawn_gazebo_robot.launch" >
    <arg name="robot" value="audibot" />
    <arg name="description_pkg" value="$(find audibot_description)" />
    <arg name="control_pkg" value="$(find audibot_control)" />
    <arg name="twist_control" value="false" />
    <arg name="start_x" value="-40.0" />
    <arg name="start_y" value="40.0" />
    <arg name="start_z" value="0.3" />
    <arg name="gps" value="true" />
    <arg name="gps_rate" value="50.0" />
    <arg name="pub_tf" value="1" />
    <arg name="use_sensors" value="1" />
    <arg name="pub_heading" value="1" />
    <arg name="tf_freq" value="1000.0" />
  </include>

  <param name="/audibot/gps/ref_lat" value="42.853452" />
  <param name="/audibot/gps/ref_lon" value="-83.069858" />

<!-- World launch -->
  <include file="$(find ugv_course_gazebo)/launch/gazebo_world.launch" >
    <arg name="world_name" value="$(find spot_detection)/world/parking1.world" />
    <arg name="headless" value="false" />
  </include>

<!-- lane_extract node (for parameter settings)
  <node pkg="spot_detection" type="lane_extract" name="lane_extract" output="screen" >
    <param name="hough_threshold" value="25" />
    <param name="hough_min_line" value="100" />
    <param name="hough_max_line" value="30" />
  </node>
-->

<!-- Vehicle Control Node --> 
  <node pkg="spot_detection" type="vehicle_control" name="vehicle_control" output="screen" />


<!-- Find Parking Node 
  <node pkg="spot_detection" type="find_parking" name="find_parking" output="screen" >
    <param name="hough_threshold" value="25" />
    <param name="hough_min_line" value="100" />
    <param name="hough_max_line" value="30" />
    <param name="h_center" value="30" />
    <param name="h_width" value="10" />
  </node>
-->

  <!-- Odom Node -->
  <node pkg="bicycle_state_space" type="bicycle_state_space" name="bicycle_state_space" output="screen" >
    <remap from="/twist" to="/audibot/cmd_vel" />
  </node>
  
  <!-- Static Transfrom -->
  <node pkg="tf" type="static_transform_publisher" name="world_to_map" args="0 0 0 0 0 0 /world /map 30" />
  
  
</launch>
